server/src/test/java/com/objectcomputing/checkins/services/memberprofile/MemberProfileControllerTest.java

```diff
@@ -9,277 +11,263 @@
     @Test
-    public void testGetFindByIdReturns404() {
+    public void testGETFindByValueName() {
 
-        when(mockMemberServices.getById(testUuid))
-                .thenThrow(new MemberProfileDoesNotExistException("testMessage"));
+        MemberProfile memberProfile = createADefaultMemberProfile();
+        final HttpRequest<Object> request = HttpRequest.
+                GET(String.format("/?name=%s", encodeValue(memberProfile.getName()))).basicAuth(MEMBER_ROLE,MEMBER_ROLE);
 
-        HttpClientResponseException thrown = assertThrows(HttpClientResponseException.class, () -> {
-            client.toBlocking().exchange(HttpRequest
-                    .GET(String.format("/%s", testUuid))
-                    .basicAuth(MEMBER_ROLE, MEMBER_ROLE));
-        });
-        assertEquals("testMessage", thrown.getMessage());
-        assertEquals(HttpStatus.NOT_FOUND, thrown.getStatus());
+        final HttpResponse<Set<MemberProfile>> response = client.toBlocking().exchange(request, Argument.setOf(MemberProfile.class));
+
+        assertEquals(Set.of(memberProfile), response.body());
+        assertEquals(HttpStatus.OK,response.getStatus());
     }
 
-    // Find By Name - when no user data exists
     @Test
-    public void testGetFindByNameReturnsEmptyBody() {
+    public void testGETGetByIdHappyPath() {
 
-        MemberProfile memberProfile = mkMemberProfile();
-        memberProfile.setName(memberProfile.getName());
+        MemberProfile memberProfile = createADefaultMemberProfile();
 
-        when(mockMemberServices.findByValues(memberProfile.getName(), null, null, null))
-                .thenReturn(Collections.EMPTY_SET);
+        final HttpRequest<Object> request = HttpRequest.
+                GET(String.format("/%s", memberProfile.getId())).basicAuth(MEMBER_ROLE,MEMBER_ROLE);
 
-        HttpRequest request = HttpRequest.GET(String.format("/?name=%s", testUser))
-                .basicAuth(MEMBER_ROLE, MEMBER_ROLE);
-        List<MemberProfileResponseDTO> response = client.toBlocking().retrieve(request, Argument.listOf(MemberProfileResponseDTO.class));
+        final HttpResponse<MemberProfile> response = client.toBlocking().exchange(request, MemberProfile.class);
 
-        assertEquals(0, response.size());
+        assertEquals(memberProfile, response.body());
+        assertEquals(HttpStatus.OK,response.getStatus());
     }
 
-    // Find By Role - when no user data exists
     @Test
-    public void testGetFindByRoleReturnsEmptyBody() {
-
-        MemberProfile memberProfile = mkMemberProfile();
-        memberProfile.setName(memberProfile.getName());
+    public void testGETGetByIdNotFound() {
 
-        when(mockMemberServices.findByValues(null, memberProfile.getTitle(), null, null))
-                .thenReturn(Collections.EMPTY_SET);
+        final HttpRequest<Object> request = HttpRequest.
+                GET(String.format("/%s", UUID.randomUUID().toString())).basicAuth(MEMBER_ROLE,MEMBER_ROLE);
```

Comment:
I know the answer is "really really close to 0" but I'd be interested to know what the odds are this could coincidentally generate a UUID that's in the database and cause the test to fail.

Reply:
The number of random version-4 UUIDs which need to be generated in order to have a 50% probability of at least one collision is 2.71 quintillion. 

1 Quintillion = 1 followed by 18 zeros =)

https://en.wikipedia.org/wiki/Universally_unique_identifier#Collisions
https://stackoverflow.com/questions/2513573/how-good-is-javas-uuid-randomuuid

Reply:
![So you are telling me there is a chance...](https://media.tenor.com/images/e00ff448e4f495a435938e6130466cd3/tenor.gif)

Reply:
https://gph.is/g/apkp2D6

Reply:
https://media.giphy.com/media/fqHkeAzgD437O/giphy.gif

Reply:
![ygbkm](https://user-images.githubusercontent.com/48927697/99454720-6e214880-28ec-11eb-9f7c-7e763e8d8e9f.gif)